Lib/idlelib/macosx.py

```diff
@@ -33,17 +34,20 @@ def _init_tk_type():
                 _tk_type = "cocoa"
                 return
 
-        root = tkinter.Tk()
-        ws = root.tk.call('tk', 'windowingsystem')
+            else:
+                if _idle_root is None:
+                    _tk_type = "cocoa"
+                    return
```

Comment:
I believe everything from the 'testing' import to here can be deleted and these 3 lines dedented twice.  IDLE will set _idle_root, at least when on mac and if a test on mac needs an accurate _tk_type, it should check requires('gui') and set _idle_root.  Checking it here is redundant.  Do as you wish for now.

Reply:
I'm leaving this as is for now, I can check later if it is save to remove the testing import.